Lib/xmlrpc/client.py

```diff
@@ -245,41 +245,12 @@ def __repr__(self):
 
 ##
 # Backwards compatibility
-
 boolean = Boolean = bool
 
-##
-# Wrapper for XML-RPC DateTime values.  This converts a time value to
-# the format used by XML-RPC.
-# <p>
-# The value can be given as a datetime object, as a string in the
-# format "yyyymmddThh:mm:ss", as a 9-item time tuple (as returned by
-# time.localtime()), or an integer value (as returned by time.time()).
-# The wrapper uses time.localtime() to convert an integer to a time
-# tuple.
-#
-# @param value The time, given as a datetime object, an ISO 8601 string,
-#              a time tuple, or an integer time value.
-
 
-# Issue #13305: different format codes across platforms
-_day0 = datetime(1, 1, 1)
-def _try(fmt):
-    try:
-        return _day0.strftime(fmt) == '0001'
-    except ValueError:
-        return False
-if _try('%Y'):      # Mac OS X
-    def _iso8601_format(value):
-        return value.strftime("%Y%m%dT%H:%M:%S")
-elif _try('%4Y'):   # Linux
-    def _iso8601_format(value):
-        return value.strftime("%4Y%m%dT%H:%M:%S")
-else:
-    def _iso8601_format(value):
-        return value.strftime("%Y%m%dT%H:%M:%S").zfill(17)
-del _day0
-del _try
+def _iso8601_format(value):
+    # XML-RPC doesn't use '-' separator in the date part
+    return value.isoformat(timespec='seconds').replace('-', '')
```

Comment:
Might be worth a comment indicating that `value` must be a naïve `datetime` here, since `-` is also used to indicate negative UTC offsets, e.g.:

```python
>>> from datetime import datetime
>>> from zoneinfo import ZoneInfo
>>> dt = datetime(2023, 6, 12, 13, 30, tzinfo=ZoneInfo("America/New_York"))
>>> print(dt.isoformat(timespec="seconds"))
2023-06-12T13:30:00-04:00
```

The function you are replacing ignores the time zone information, so if there's some possibility that `value` might be time zone aware, a more equivalent function would be this one:

```python
def _iso8601_format(value):
    if value.tzinfo is not None:
        # XML RPC only uses the naïve portion of the datetime
        value = value.replace(tzinfo=None)
    return value.isoformat(timespec="seconds").replace("-", "")
```

Reply:
Oh. I have no idea if aware datetime objects are accepted or not: there are no tests for them :-) I added a test and your proposed change, so my change simply keeps the old behavior.